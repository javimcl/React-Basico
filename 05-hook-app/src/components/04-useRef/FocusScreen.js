import React, { useRef } from 'react';
import '../02-useEffect/effect.css'

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick=()=> {
       // document.querySelector('input').focus();
       inputRef.current.select();
    }
    return (
        <div>
            <h1>Focus</h1>
            <hr></hr>
            <input ref={inputRef}
            className='form-control'
            placeholder='Su nombre'>
            </input>
            <button 
            className='btn btn-outline-primary mt-5'
            onClick={handleClick}>
                Focus
            </button>
            
        </div>
    )
}
