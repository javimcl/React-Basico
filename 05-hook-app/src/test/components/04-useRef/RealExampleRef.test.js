import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";


describe('Pruenas en <RealExample/>', () => {
  test('debe mostrarse correctamente', () => {
      const wrapper = shallow(<RealExampleRef/>)

      expect(wrapper).toMatchSnapshot();
    
  });

  test('debe de mostrar el componente <multipleCusmo/>', () => {
    
  });
  
  
});
