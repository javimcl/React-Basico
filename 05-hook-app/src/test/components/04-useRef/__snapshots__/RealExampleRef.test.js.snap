// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruenas en <RealExample/> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
